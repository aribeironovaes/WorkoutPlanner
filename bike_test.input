(10min, 85, HIM)
-- EOT
(20min, 55, 75, Warm up)
-- EOT
4[(45s, 75, 100), (15s, 55)]
-- EOT
3[(1min, 80, 90, 100), (30s, 50)]
-- EOT
(5min, easy)
-- EOT
(10min, 100, FTP, 2min)
-- EOT
