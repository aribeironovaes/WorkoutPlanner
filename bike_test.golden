Input: 
(1hr, 75)
Normalized: 
(1hr, 75)
AST: 
ArrayInterval(
	SimpleInterval(1hr, 75%, )
)
IF (Avg): 0.75
TSS: 56.3
TSS2: 60
Intensity Unit: %
Pretty Print:
	* 1hr @ 75%
Duration Unit: h
Duration (Sec): 3600
----------------------------
Input: 
(1hr, 85)
Normalized: 
(1hr, 85)
AST: 
ArrayInterval(
	SimpleInterval(1hr, 85%, )
)
IF (Avg): 0.85
TSS: 72.2
TSS2: 76.5
Intensity Unit: %
Pretty Print:
	* 1hr @ 85%
Duration Unit: h
Duration (Sec): 3600
----------------------------
Input: 
(1min, 85)
Normalized: 
(1min, 85)
AST: 
ArrayInterval(
	SimpleInterval(1min, 85%, )
)
IF (Avg): 0.85
TSS: 1.2
TSS2: 1.3
Intensity Unit: %
Pretty Print:
	* 1' @ 85%
Duration Unit: min
Duration (Sec): 60
----------------------------
Input: 
(1min, 60, 80)
Normalized: 
(1min, 60, 80)
AST: 
ArrayInterval(
	BuildInterval(1min, 60%, 80%, )
)
IF (Avg): 0.7071067811865476
TSS: 0.8
TSS2: 0.8
Intensity Unit: %
Pretty Print:
	* 1' build from 60% to 80%
Duration Unit: min
Duration (Sec): 60
----------------------------
Input: 
(1s, 50), (2s, 155w), (3s, 50%)
Normalized: 
(1sec, 50)
(2sec, 50)
(3sec, 50)
AST: 
ArrayInterval(
	SimpleInterval(1sec, 50%, )
	SimpleInterval(2sec, 155w, )
	SimpleInterval(3sec, 50%, )
)
IF (Avg): 0.5
TSS: 0
TSS2: 0
Duration Unit: s
Duration (Sec): 6
----------------------------
Input: 
(1hr, 75), (1hr, 85)
Normalized: 
(1hr, 75)
(1hr, 85)
AST: 
ArrayInterval(
	SimpleInterval(1hr, 75%, )
	SimpleInterval(1hr, 85%, )
)
IF (Avg): 0.8015609770940698
TSS: 128.5
TSS2: 128.2
Intensity Unit: %
Pretty Print:
	* 1hr @ 75%
	* 1hr @ 85%
Duration Unit: h
Duration (Sec): 7200
----------------------------
Input: 
(1min, 55, 75)
Normalized: 
(1min, 55, 75)
AST: 
ArrayInterval(
	BuildInterval(1min, 55%, 75%, )
)
IF (Avg): 0.6576473218982952
TSS: 0.7
TSS2: 0.8
Intensity Unit: %
Pretty Print:
	* 1' warm up to 75%
Duration Unit: min
Duration (Sec): 60
----------------------------
Input: 
(10min, 85, HIM)
Normalized: 
(10min, 85, HIM)
AST: 
ArrayInterval(
	SimpleInterval(10min, 85%, HIM)
)
IF (Avg): 0.85
TSS: 12
TSS2: 12.8
Intensity Unit: %
Pretty Print:
	* 10' HIM @ 85%
Duration Unit: min
Duration (Sec): 600
----------------------------
Input: 
(20min, 55, 75, Warm up)
Normalized: 
(20min, 55, 75, Warm up)
AST: 
ArrayInterval(
	BuildInterval(20min, 55%, 75%, Warm up)
)
IF (Avg): 0.6576473218982952
TSS: 14.2
TSS2: 15.3
Intensity Unit: %
Pretty Print:
	* 20' warm up to 75%
Duration Unit: min
Duration (Sec): 1200
----------------------------
Input: 
4[(45s, 75, 100), (15s, 55)]
Normalized: 
4[(45sec, 75, 100), (15sec, 55)]
AST: 
ArrayInterval(
	RepeatInterval(count=4, 
		BuildInterval(45sec, 75%, 100%, )
		SimpleInterval(15sec, 55%, )
	)
)
IF (Avg): 0.8133649242498721
TSS: 4.3
TSS2: 4.3
Intensity Unit: %
Pretty Print:
	* 4 x 45'' build from 75% to 100% - 15'' easy
Duration Unit: s
Duration (Sec): 240
----------------------------
Input: 
3[(1min, 80, 90, 100), (30s, 50)]
Normalized: 
3[(1min, 80, 90, 100), (30sec, 50)]
AST: 
ArrayInterval(
	)
IF (Avg): 0.8577378887016043
TSS: 4.7
TSS2: 5.1
Intensity Unit: %
Pretty Print:
	* 3 x 1' - w/ 30'' easy (80%, 90%, 100%)
Duration (Sec): 270
----------------------------
Input: 
(5min, easy)
Normalized: 
(5min, 55, easy)
AST: 
ArrayInterval(
	SimpleInterval(5min, 55%, easy)
)
IF (Avg): 0.55
TSS: 2.5
TSS2: 2.8
Intensity Unit: %
Pretty Print:
	* 5' easy @ 55%
Duration Unit: min
Duration (Sec): 300
----------------------------
Input: 
(10min, 100, FTP, 2min)
Normalized: 
(10min, 100, FTP, 2min)
AST: 
ArrayInterval(
	SimpleInterval(10min, 100%, FTP, 2min)
)
IF (Avg): 1
TSS: 16.7
TSS2: 20
Intensity Unit: %
Pretty Print:
	* 10' FTP @ 100% w/ 2' rest
Duration Unit: min
Duration (Sec): 720
----------------------------
Input: 
(1min, 55, 70rpm)
Normalized: 
(1min, 55, 70rpm)
AST: 
ArrayInterval(
	SimpleInterval(1min, 55%, 70rpm)
)
IF (Avg): 0.55
TSS: 0.5
TSS2: 0.6
Intensity Unit: %
Pretty Print:
	* 1' 70rpm @ 55%
Duration Unit: min
Duration (Sec): 60
----------------------------
Input: 
(1min, 55, 2% grade)
Normalized: 
(1min, 55, 2% grade)
AST: 
ArrayInterval(
	SimpleInterval(1min, 55%, 2% grade)
)
IF (Avg): 0.55
TSS: 0.5
TSS2: 0.6
Intensity Unit: %
Pretty Print:
	* 1' 2% grade @ 55%
Duration Unit: min
Duration (Sec): 60
----------------------------
Input: 
(1hr, 75), (20mi, 85)
Normalized: 
(1hr, 75)
(20mi, 85)
AST: 
ArrayInterval(
	SimpleInterval(1hr, 75%, )
	SimpleInterval(20mi, 85%, )
)
IF (Avg): 0.8015609770940698
TSS: 128.5
TSS2: 128.2
Intensity Unit: %
Pretty Print:
	* 1hr @ 75%
	* 20mi @ 85%
Duration (Sec): 7200
----------------------------
Input: 
(10min, *, Free)
Normalized: 
(10min, *, Free)
AST: 
ArrayInterval(
	SimpleInterval(10min, *, Free)
)
IF (Avg): 0.55
TSS: 5
TSS2: 5.5
Intensity Unit: free-ride
Pretty Print:
	* 10' Free
Duration Unit: min
Duration (Sec): 600
----------------------------
Input: 
(10min, 100, FTP)
Normalized: 
(10min, 100, FTP)
AST: 
ArrayInterval(
	SimpleInterval(10min, 100%, FTP)
)
IF (Avg): 1
TSS: 16.7
TSS2: 16.7
Intensity Unit: %
Pretty Print:
	* 10' FTP @ 100%
Duration Unit: min
Duration (Sec): 600
----------------------------
Input: 
(10sec, *)
Normalized: 
(10sec, *)
AST: 
ArrayInterval(
	SimpleInterval(10sec, *, )
)
IF (Avg): 0.55
TSS: 0.1
TSS2: 0.1
Intensity Unit: free-ride
Pretty Print:
	* 10''
Duration Unit: s
Duration (Sec): 10
----------------------------
Input: 
2[(1min, 85, 95), (30sec, 50)]
Normalized: 
2[(1min, 85, 95), (30sec, 50)]
AST: 
ArrayInterval(
	)
IF (Avg): 0.8366600265340756
TSS: 3.1
TSS2: 3.3
Intensity Unit: %
Pretty Print:
	* 2 x 1' - w/ 30'' easy (85%, 95%)
Duration (Sec): 180
----------------------------
Input: 
2[(100, 30sec, 45sec), (30sec, 50)]
Normalized: 
2[(100, 30sec, 45sec), (30sec, 50)]
AST: 
ArrayInterval(
	)
IF (Avg): 0.8864052604279183
TSS: 2.5
TSS2: 3
Intensity Unit: %
Pretty Print:
	* 2 x 100% - w/ 30'' easy (30'', 45'')
Duration Unit: s
Duration (Sec): 135
----------------------------
Input: 
(10min, 100, FTP), (5min, 55, easy)
Normalized: 
(10min, 100, FTP)
(5min, 55, easy)
AST: 
ArrayInterval(
	SimpleInterval(10min, 100%, FTP)
	SimpleInterval(5min, 55%, easy)
)
IF (Avg): 0.8760707733967615
TSS: 19.2
TSS2: 19.7
Intensity Unit: %
Pretty Print:
	* 10' FTP @ 100%
	* 5' easy @ 55%
Duration Unit: min
Duration (Sec): 900
----------------------------
Input: 
4[(1min, 100, hard), (30sec, 50, easy)]
Normalized: 
4[(1min, 100, hard), (30sec, 50, easy)]
AST: 
ArrayInterval(
	RepeatInterval(count=4, 
		SimpleInterval(1min, 100%, hard)
		SimpleInterval(30sec, 50%, easy)
	)
)
IF (Avg): 0.8660254037844386
TSS: 7.5
TSS2: 7.8
Intensity Unit: %
Pretty Print:
	* 4 x 1' hard @ 100% - 30'' easy
Duration (Sec): 360
----------------------------
Input: 
(155w, 310w, 1min, build to ftp)
Normalized: 
(1min, 50, 100, build to ftp)
AST: 
ArrayInterval(
	BuildInterval(1min, 155w, 310w, build to ftp)
)
IF (Avg): 0.7905694150420949
TSS: 1
TSS2: 1.1
Intensity Unit: w
Pretty Print:
	* 1' warm up to 310w
Duration Unit: min
Duration (Sec): 60
----------------------------
Input: 
(10min)
Normalized: 
(10min, 55)
AST: 
ArrayInterval(
	SimpleInterval(10min, 55%, )
)
IF (Avg): 0.55
TSS: 5
TSS2: 5.5
Intensity Unit: %
Pretty Print:
	* 10' easy
Duration Unit: min
Duration (Sec): 600
----------------------------
Input: 
(10min, easy)
Normalized: 
(10min, 55, easy)
AST: 
ArrayInterval(
	SimpleInterval(10min, 55%, easy)
)
IF (Avg): 0.55
TSS: 5
TSS2: 5.5
Intensity Unit: %
Pretty Print:
	* 10' easy @ 55%
Duration Unit: min
Duration (Sec): 600
----------------------------
Input: 
(10s, *, free ride)
Normalized: 
(10sec, *, free ride)
AST: 
ArrayInterval(
	SimpleInterval(10sec, *, free ride)
)
IF (Avg): 0.55
TSS: 0.1
TSS2: 0.1
Intensity Unit: free-ride
Pretty Print:
	* 10'' free ride
Duration Unit: s
Duration (Sec): 10
----------------------------
Input: 
1[(1hr, 85)]
Normalized: 
1[(1hr, 85)]
AST: 
ArrayInterval(
	RepeatInterval(count=1, 
		SimpleInterval(1hr, 85%, )
	)
)
IF (Avg): 0.85
TSS: 72.2
TSS2: 76.5
Intensity Unit: %
Pretty Print:
	* 1 x 1hr @ 85%
Duration Unit: h
Duration (Sec): 3600
----------------------------
Input: 
2[(1hr, 85)]
Normalized: 
2[(1hr, 85)]
AST: 
ArrayInterval(
	RepeatInterval(count=2, 
		SimpleInterval(1hr, 85%, )
	)
)
IF (Avg): 0.85
TSS: 144.5
TSS2: 153
Intensity Unit: %
Pretty Print:
	* 2 x 1hr @ 85%
Duration Unit: h
Duration (Sec): 7200
----------------------------
Input: 
(1hr, 75) 2[(1hr, 85)]
Normalized: 
(1hr, 75)
2[(1hr, 85)]
AST: 
ArrayInterval(
	SimpleInterval(1hr, 75%, )
	RepeatInterval(count=2, 
		SimpleInterval(1hr, 85%, )
	)
)
IF (Avg): 0.8180260794538684
TSS: 200.8
TSS2: 196.3
Intensity Unit: %
Pretty Print:
	* 1hr @ 75%
	* 2 x 1hr @ 85%
Duration Unit: h
Duration (Sec): 10800
----------------------------
Input: 
(1hr, 86)
Normalized: 
(1hr, 86)
AST: 
ArrayInterval(
	SimpleInterval(1hr, 86%, )
)
IF (Avg): 0.86
TSS: 74
TSS2: 77.4
Intensity Unit: %
Pretty Print:
	* 1hr @ 86%
Duration Unit: h
Duration (Sec): 3600
----------------------------
Input: 
(30min, 60), (30min, 100)
Normalized: 
(30min, 60)
(30min, 100)
AST: 
ArrayInterval(
	SimpleInterval(30min, 60%, )
	SimpleInterval(30min, 100%, )
)
IF (Avg): 0.8246211251235321
TSS: 68
TSS2: 74.2
Intensity Unit: %
Pretty Print:
	* 30' @ 60%
	* 30' @ 100%
Duration Unit: min
Duration (Sec): 3600
----------------------------
Input: 
4[(45s, 75, 100), (15s, 55)]
Normalized: 
4[(45sec, 75, 100), (15sec, 55)]
AST: 
ArrayInterval(
	RepeatInterval(count=4, 
		BuildInterval(45sec, 75%, 100%, )
		SimpleInterval(15sec, 55%, )
	)
)
IF (Avg): 0.8133649242498721
TSS: 4.3
TSS2: 4.3
Intensity Unit: %
Pretty Print:
	* 4 x 45'' build from 75% to 100% - 15'' easy
Duration Unit: s
Duration (Sec): 240
----------------------------
Input: 
3[(2min, 1min, 1min, 75), (1min, 55)]
Normalized: 
3[(75, 2min, 1min, 1min), (1min, 55)]
AST: 
ArrayInterval(
	)
IF (Avg): 0.71449282711585
TSS: 5.3
TSS2: 5.8
Intensity Unit: %
Pretty Print:
	* 3 x 75% - w/ 1' easy (2', 1', 1')
Duration Unit: min
Duration (Sec): 420
----------------------------
Input: 
3[(1min, 80, 90, 100), (30s, 50)]
Normalized: 
3[(1min, 80, 90, 100), (30sec, 50)]
AST: 
ArrayInterval(
	)
IF (Avg): 0.8577378887016043
TSS: 4.7
TSS2: 5.1
Intensity Unit: %
Pretty Print:
	* 3 x 1' - w/ 30'' easy (80%, 90%, 100%)
Duration (Sec): 270
----------------------------
Input: 
(1:30min, 55)
Normalized: 
(1:30min, 55)
AST: 
ArrayInterval(
	SimpleInterval(1:30min, 55%, )
)
IF (Avg): 0.55
TSS: 0.8
TSS2: 0.8
Intensity Unit: %
Pretty Print:
	* 1'30'' easy
Duration Unit: min
Duration (Sec): 90
----------------------------
